### ISSUE TITLE
Failed to load external drawtext plugin

### ISSUE THREAD

**Issue by tjtalkar** [Psychtoolbox-3/Psychtoolbox-3#583]
Hi,

I've been through issues #429 and #579 regarding the DrawFormattedText on Windows. Specifically, the error is "Failed to load external drawtext plugin 'libptbdrawtext_ftgl64.dll'. Reverting to legacy GDI text renderer". This error causes any text from DrawFormattedText to show up as cutoff on top with incorrect line spacing. I've cleared the fontconfig folder, waited, and also reinstalled gstreamer and run as administrator. I'm not sure if there's anything else settings wise that can be changed - I've checked that the file is present in the folder and accessible by Matlab on my machine.

Thanks!

**Comment by dcnieho** [comment]
Is your ptb up to date (3.0.16) and did you install the exact right version
of gstreamer?

**Comment by kleinerm** [comment]
That's not related to any of these issues, but maybe Matlab ships an outdated libfreetype.dll, or an UpdatePsychtoolbox will make sure you have the latest version of the drawtext plugin, see all tips in
http://psychtoolbox.org/docs/DrawTextPlugin

**Comment by jantheman** [comment]
Same issue here.

Installation/Update of toolbox reports:
```matlab
Warning: Something went wrong with high-quality text renderer setup. Read 'help DrawTextPlugin' for troubleshooting. 
> In PsychtoolboxPostInstallRoutine (line 735)
  In UpdatePsychtoolbox (line 305) 
```

Calling DrawText reports:
```
PTB-DEBUG: DrawText: Trying to load external text renderer plugin from following file: [ C:\D\Psychtoolbox\PsychBasic\PsychPlugins\libptbdrawtext_ftgl64.dll ]
PTB-DEBUG: DrawText: Failed to load external drawtext plugin [Unknown error].
PTB-INFO: DrawText: Failed to load external drawtext plugin 'libptbdrawtext_ftgl64.dll'. Reverting to legacy GDI text renderer. 'help DrawTextPlugin' for troubleshooting.
```

Windows version: `Windows 10 21H2 (10.0.19044.2965)`
Matlab version: `2020a Update 5`
Toolbox version: `3.0.19 - Flavor: beta - Corresponds to SVN Revision 13692`
GStreamer version: `1.22.3`
MSVC version: `14.28.29914.0`

No _fontconfig_ folder gets created.

Renaming _C:\Program Files\MATLAB\R2020a\bin\win64\libfreetype.dll_ had no effect.

Any troubleshooting/diagnosis instructions I could follow?

**Comment by jantheman** [comment]
I suggest the call to `LoadLibrary` which is trying to load the `libptbdrawtext_ftgl64.dll` is followed by `GetLastError` and the error is reported to the user instead of 'Unknown error'.

This returned `The specified module could not be found` to me although the file exists at the requested path. Further trace revealed other libraries missing:

![image](https://github.com/Psychtoolbox-3/Psychtoolbox-3/assets/10547386/ff8381dc-782d-46a9-b9cf-afbd7ef3bbed)

There is _C:\Program Files\MATLAB\R2020a\bin\win64\libfreetype.dll_ instead and no _fontconfig_ dll at all.

**Comment by dcnieho** [comment]
I've seen the same issue on two devices recently, haven't had time to
report yet. Latest GStreamer doesn't work. The minimal compatible one
(1.20.5) does. Don't install the latest but the msvc version of this older
one

**Comment by kleinerm** [comment]
@dcnieho GStreamer 1.22.1 was tested and working on my Windows 10 22H2 with R2022b, that's why it is in the help text.

@jantheman Well tested pull request for improved error reporting welcome. But it is almost always "specified module could not be found" which usually means some required dll (fontconfig / freetype) was not found. Maybe some path problem with GStreamer? Anyhow, `help PsychPaidSupportAndServices` for a paid support authentication token, if you want 30 minutes of my work time for any further advice. I've spent days fixing this shitshow on Windows for the 3.0.19 release, in the hope many of the text rendering problems would go away on Windows. I can't afford to spend more free work time fixing up Windows again.

**Comment by kleinerm** [comment]
This should be solved by a new drawtext plugin in the next 3.0.19.4 release. Won't work on GStreamer versions older than 1.22 from then on, but the current 1.22 series should work. See:

https://psychtoolbox.discourse.group/t/gstreamer-drawtext-trouble-on-ms-windows-was-movies-not-loading/5046/7

Financial loss to the PTB project due to nobody paying for this: Over 3 hours of billable work time.
